client/client_test.go

```diff
@@ -2213,6 +2215,92 @@ func testTarExporterSymlink(t *testing.T, sb integration.Sandbox) {
 	require.Equal(t, "foo", item.Header.Linkname)
 }
 
+func testBuildExportWithForeignLayer(t *testing.T, sb integration.Sandbox) {
+	if os.Getenv("TEST_DOCKERD") == "1" {
+		t.Skip("image exporter is missing in dockerd")
+	}
+
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	st := llb.Image("cpuguy83/buildkit-foreign:latest")
+	def, err := st.Marshal(sb.Context())
+	require.NoError(t, err)
+
+	registry, err := sb.NewRegistry()
+	if errors.Is(err, integration.ErrRequirements) {
+		t.Skip(err.Error())
+	}
+	require.NoError(t, err)
+
+	t.Run("propagate=1", func(t *testing.T) {
+		target := registry + "/buildkit/build/exporter:withforeignlayer"
+		_, err = c.Solve(sb.Context(), def, SolveOpt{
+			Exports: []ExportEntry{
+				{
+					Type: ExporterImage,
+					Attrs: map[string]string{
+						"name":                  target,
+						"push":                  "true",
```

Comment:
Not very important if it is tricky but is there a simple command we could run against the registry to check that it didn't push the foreign layer.

Reply:
Added code to check the registry.